src/components/ciPipeline/WebhookCD/WebhookDetails.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useRef } from 'react'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+import { Drawer} from '../../common'
+import { ReactComponent as Help } from '../../../assets/icons/ic-help.svg'
+import { ReactComponent as CopyIcon } from '../../../assets/icons/ic-copy.svg'
+import { ReactComponent as InfoIcon } from '../../../assets/icons/info-filled.svg'
+import InfoColourBar from '../../common/infocolourBar/InfoColourbar'
+import { CIPipelineType } from '../types'
+import './webhookDetails.scss'
+
+export function WebhookDetails({ appName, connectCDPipelines, getWorkflows, close, deleteWorkflow }: CIPipelineType) {
+    const appStatusDetailRef = useRef<HTMLDivElement>(null)
+    const escKeyPressHandler = (evt): void => {
+        if (evt && evt.key === 'Escape' && typeof close === 'function') {
+            evt.preventDefault()
+            close()
+        }
+    }
+
+    const closeWebhook = ():void=>{
+      close()
+    }
+
+    useEffect(() => {
+        document.addEventListener('keydown', escKeyPressHandler)
+        return (): void => {
+            document.removeEventListener('keydown', escKeyPressHandler)
+        }
+    }, [escKeyPressHandler])
+    return (
+        <Drawer position="right" width="1024px">
```

Comment:
If this is a Drawer we can extend the name of the component as WebhookDetailsModal

Reply:
Will do it in next PR